### PR TITLE
swaps: settings e2e

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes TEAM2-429 TEAM2-430
Figma link (if any):

## What changed (plus any additional context for devs)
handles all settings functionality and currency selection network switching

toggling flashbots, adjusting route, editing slippage, & resetting to defaults 

also renamed some callbacks for readability

## What to test

green = gucci

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: e2e/swapSheetFlow.spec.js [hunk 1]
```diff
@@ -241,6 +255,14 @@ describe('Swap Sheet Interaction Flow', () => {
     await Helpers.clearField('exchange-modal-input-native-246');
     await Helpers.checkIfVisible('exchange-modal-input');
     await Helpers.checkIfVisible('exchange-modal-output');
+  });
+
+  it('Should not show settings routes picker on Arbitrum or flashbots on L2s', async () => {
+    await Helpers.waitAndTap('exchange-settings-button');
+    await Helpers.checkIfVisible('swap-settings-header');
+    await Helpers.checkIfNotVisible('swap-settings-routes-label');
+    await Helpers.checkIfNotVisible('swap-settings-flashbots-label');
```
Comment (estebanmino) [inline e2e/swapSheetFlow.spec.js:264]: this is nice, could we also add checks for flashbots not visible on optimism and polygon as well?

File: e2e/swapSheetFlow.spec.js [hunk 2]
```diff
@@ -421,6 +443,27 @@ describe('Swap Sheet Interaction Flow', () => {
...
+    await Helpers.clearField('swap-slippage-input-2');
+    await Helpers.typeText('swap-slippage-input-', '10', false);
+    await Helpers.checkIfVisible('swap-slippage-input-10');
+    await Helpers.waitAndTap('swap-settings-defaults-button');
+    await Helpers.checkIfVisible('swap-settings-routes-current-rainbow');
+    await Helpers.checkIfVisible('swap-settings-flashbots-switch-false');
+    await Helpers.checkIfVisible('swap-slippage-input-2');
+    await Helpers.waitAndTap('swap-slippage-label');
+    await Helpers.checkIfVisible('explain-sheet-slippage');
+    await Helpers.swipe('explain-sheet-slippage', 'down', 'slow');
```
Comment (estebanmino) [inline e2e/swapSheetFlow.spec.js:466]: looking nice, could you update the name of this test to say this is happening on mainnet? it wasn't clear to me until i saw it running

File: e2e/swapSheetFlow.spec.js [hunk 3]
```diff
@@ -70,11 +70,25 @@ describe('Swap Sheet Interaction Flow', () => {
...
+    await Helpers.tapByText('Optimism');
+    await Helpers.waitAndTap('currency-select-network-switcher-10');
+    await Helpers.tapByText('Arbitrum');
+    await Helpers.waitAndTap('currency-select-network-switcher-42161');
+    await Helpers.tapByText('Polygon (Matic)');
+    await Helpers.checkIfVisible('currency-select-network-switcher-137');
+    await Helpers.waitAndTap('currency-select-header-back-button');
+    await Helpers.tap('exchange-modal-output-selection-button');
+    await Helpers.checkIfVisible('currency-select-list');
+    await Helpers.checkIfVisible('currency-select-network-switcher-1');
```
Comment (estebanmino) [inline e2e/swapSheetFlow.spec.js:88]: 👌

**Review by estebanmino — APPROVED** [review]
🐐

**Review by derHowie — APPROVED** [review]
lgtm + 🧪 🐐

**Pull request merged**